src/components/playback/playback-control.tsx

```diff
@@ -0,0 +1,189 @@
+                                  if (playing) {
+                                    setSliderPlaying(playing);
+                                  } else {
+                                    setSliderPlaying(!sliderPlaying);
+                                  }
+                                },[sliderPlaying]);
+  useEffect(() => {
+    if (sliderPlaying) {
+      const slider = setInterval(()=>{
+        if (sliderValue < maxValue) {
+          setSliderValue(sliderValue + 1);
+          treeManager.setCurrentHistoryIndex(sliderValue);
+          treeManager.getHistoryEntry(sliderValue);
+        } else {
+          handlePlayPauseToggle(false);
+        }
+      }, 1000);
+      return () => clearInterval(slider);
+    }
+  }, [handlePlayPauseToggle, history.length, maxValue, sliderPlaying, sliderValue, treeManager]);
+
+
+  //TODO: need to add a modal that warns users about max number of markers. Currently, a generic alert is shown
+  //TODO: Currently, if add marker is on and user moves the time handle, a marker is added where the user
+  // drops the handle. It shouldn't do that. Marker should only be added when user clicks on the rail or the time thumb.
+  // TODO: Marker toolbar should be hidden until thumbnail is focused.
+  //        Currently, toolbar appears when user opens playback.
+  const addMarker = (value: any) => {
+    if (markers.length >= 9) {
+      alert("You already have 9 markers. Please delete some markers to add more.");
+    } else {
+      setMarkers([...markers, {id: markers.length+1, location: value}]);
+    }
+  };
+
+  //TODO: need to track which marker is selected. Currently, all markers in a document will appear selected.
+  const handleMarkerSelected = (e:  React.MouseEvent) => {
+    setMarkerSelected(!markerSelected);
+  };
+
+  const handleAddMarkerButtonSelected = () => {
+    setAddMarkerButtonSelected(!addMarkerButtonSelected);
+  };
+
+  const handleSliderValueChange = (value: any) => {
+    setSliderValue(value);
+    treeManager.setCurrentHistoryIndex(value);
+    treeManager.getHistoryEntry(value);
+  };
+
+  const handleAddMarker = (value: any) => {
+    addMarkerButtonSelected && addMarker(value);
+  };
+
+  const renderTimeInfo = () => {
+    const monthMap: Record<number,string> = {0: "Jan", 1: "Feb", 2: "Mar", 3: "Apr", 4: "May", 5: "Jun",
+                      6: "Jul", 7: "Aug", 8: "Sep", 9: "Oct", 10: "Nov", 11: "Dec"};
+    const date = eventCreatedTime;
+    const month = date?.getMonth();
+    const monthStr = month && monthMap[month];
+    const dateDay = date?.getDate();
+    const year = date?.getFullYear();
+    let hours = date?.getHours();
+    const minutes = date?.getMinutes();
+    const ampm = hours && hours >= 12 ? 'pm' : 'am';
+    hours = hours && hours % 12;
+    hours = date ? hours : 12; // the hour '0' should be '12'
+    const minutesStr = minutes && minutes < 10 ? "0" + minutes : minutes;
+    const strTime = date ? hours + ":" + minutesStr + " " + ampm : "";
```

Comment:
Date/time formatting is a complicated area, particularly when/if localization gets involved, and we should use a library for it rather than rolling our own. Since this is already coded there's no need to change it before merging this PR but we shouldn't spend any more time on it without adopting a library.